### DISCUSSION TITLE
Bundle size / local development

### DISCUSSION THREAD

**Discussion by R-Bower** [skoropadas/ng-doc#18] [General]
Been using this extensively to build out an internal component library.  I'm loving the markdown authoring and all of the demo functionality.  Curious though – is there any way to speed up the recompile process in local development mode?  A simple markdown change can take 2-3 seconds to render.  Wondering if this is because of the large bundle, which brings me to my next point.  

My built application typically exceeds 1MB.  I only have about 10 markdown pages right now, each with varying amounts of demos (average of ~5 demos per page).  Is there a reason why the built bundle is so large?  I see about 600KB are coming from this package.  Not a dealbreaker, but worth noting.

**Comment by skoropadas** [comment]
Well, NgDoc rebuilds files and renders components/modules quite quickly, generally handling it in about ~100ms, which I consider to be a pretty good result. But, currently, it takes 700-1000ms and the problem is the delays that I set for the watcher. They work as a buffer for changed files so that the application is not rebuilt several times if multiple files are changed, but I think the delays are unreasonably large, and reducing them should speed up the rebuild and not cause any problems.

You are also partly right to mention the bundle size, which is a known problem, and I think it can also affect the rebuild speed. The bundle size increased, especially after the playground feature, and I think this is what I will focus on in the near future. I need to review dependencies and imports, and ideally, I would like to try to fit the application within default limits, but I can't promise anything.

I'll try to release a few fixes next week and see what comes out of it.

**Comment by skoropadas** [comment]
So, I played with the project a bit this weekend, I fixed some imports, reduced delays, added minification for HTML output, replaced `prettier` with `js-beautify` (I use it to format HTML in playgrounds), and did some other things, here are stats that I have after all manipulations for NgDoc documentation project:

<details>
  <summary>Before</summary>
  
<img width="352" alt="image" src="https://user-images.githubusercontent.com/20700969/221435248-6f085385-614b-48ff-933f-f3ea6505cd57.png">
  
</details>

<details>
  <summary>After</summary>
 <img width="837" alt="image" src="https://user-images.githubusercontent.com/20700969/221435144-caf67c3a-726d-47f0-b609-6298ee14140f.png">
  
</details>

Pls, try to install the 15.4.1 version and give me some feedback on how it works for your

**Reply by R-Bower** [reply]
That's good.  Looks like an excellent start.  I'll give it a shot.  Thanks for the quick action!
